src/plugins/intel_npu/src/plugin/npuw/just_sync_infer_request.cpp

```diff
@@ -168,6 +169,40 @@ ov::npuw::JustInferRequest::JustInferRequest(const std::shared_ptr<ov::npuw::Com
         m_subrequests_gio.at(sub_idx).global_results[i] = out_idx;
     }
     // }}}
+
+    // Fill weights bank if applicable
+    if (!m_npuw_model->m_weights_bank_name.empty()) {
+        // FIXME: mostly copied from unpack_closure()
+        for (size_t i = 0; i < m_num_submodels; i++) {
+            LOG_INFO("Filling weights bank for Subgraph[" << i << "]...");
+            LOG_BLOCK();
+            auto& comp_model_desc = m_npuw_model->m_compiled_submodels[i];
+
+            if (!comp_model_desc.compiled_model || (comp_model_desc.replaced_by && comp_model_desc.replaced_by != i)) {
+                OPENVINO_THROW("FATAL: weights bank encountered folded submodel!");
+            }
+
+            auto weights_bank = ov::npuw::weights::bank(m_npuw_model->m_weights_bank_name, *comp_model_desc.device_it);
+            auto& request = m_subrequests[i];
+
+            for (std::size_t cidx = 0u; cidx < comp_model_desc.closure.size(); cidx++) {
+                auto& closure = comp_model_desc.closure[cidx];
+
+                const auto closure_param_id = comp_model_desc.param_base + cidx;
+                auto& iport = comp_model_desc.compiled_model->inputs()[closure_param_id];
+                auto clparam = request->get_tensor(iport);
+                if (closure.get_element_type() == clparam->get_element_type()) {
+                    // No unpack required + utilize weights bank
+                    comp_model_desc.unpack_required[cidx] = false;
+                    closure = weights_bank->get(closure);
```

Comment:
So this is how this case needs to be handled:

1. On FOLD: The closure tensor in the closure vector should be replaced with a tensor from bank.
2. On CWAI: just call `set_tensor()` with the right tensor here and don't make it in runtime anymore.

There's problems with both cases:

1. With FOLD, we rewrite the function closure here. It was holding an original tensor but now holds its bank version. It means at one point in time we have both. So the question is - why do we have the non-bank tensor here at the first place?

2. With CWAI, we can't push the `set_tensor` early binding code down to the `CompiledModel` since there's no infer requests managed at that level. But the bank initialization should still be there. Not elegant, but the right thing to do.

Reply:
`unpack` is required if types don't match.. What we need to trace here if `set_tensor` is required in runtime even if unpack is not required. So maybe it is a matter of one more? flag vector which we'll set to false here if early binding was done (`update_required`)?

Reply:
Done